core/executor/src/native_executor.rs

```diff
@@ -59,28 +62,78 @@ pub trait NativeExecutionDispatch: Send + Sync {
 	fn native_version() -> NativeVersion;
 }
 
+/// The callable Substrate runtime. When called, a version of the Wasm runtime code is executed.
+pub trait SubstrateRuntime {
+	/// Call a method in the Substrate runtime by name. Returns the encoded result on success.
+	fn call(&mut self, ext: &mut dyn Externalities<Blake2Hasher>, method: &str, data: &[u8])
+		-> Result<Vec<u8>>;
+
+	/// Returns the version of this runtime.
+	///
+	/// Returns `None` if the runtime doesn't provide the information or there was an error
+	/// while fetching it.
+	fn version(&self) -> Option<RuntimeVersion>;
+}
+
+/// Specification of different methods of executing the runtime Wasm code.
+#[derive(Debug, PartialEq, Eq, Copy, Clone)]
+pub enum WasmExecutionMethod {
+	/// Uses the Wasmi interpreter.
+	Interpreted,
+}
+
+impl Default for WasmExecutionMethod {
+	fn default() -> Self {
+		WasmExecutionMethod::Interpreted
+	}
+}
+
 /// A generic `CodeExecutor` implementation that uses a delegate to determine wasm code equivalence
 /// and dispatch to native code when possible, falling back on `WasmExecutor` when not.
 #[derive(Debug)]
 pub struct NativeExecutor<D> {
 	/// Dummy field to avoid the compiler complaining about us not using `D`.
 	_dummy: ::std::marker::PhantomData<D>,
-	/// The fallback executor in case native isn't available.
-	fallback: WasmExecutor,
+	/// Method used to execute fallback Wasm code.
+	wasm_method: WasmExecutionMethod,
 	/// Native runtime version info.
 	native_version: NativeVersion,
 	/// The number of 64KB pages to allocate for Wasm execution.
-	default_heap_pages: Option<u64>,
+	default_heap_pages: u64,
 }
 
 impl<D: NativeExecutionDispatch> NativeExecutor<D> {
 	/// Create new instance.
-	pub fn new(default_heap_pages: Option<u64>) -> Self {
+	///
+	/// # Parameters
+	///
+	/// `wasm_method` - Method used to execute fallback Wasm code.
+	///
+	/// `default_heap_pages` - Number of 64KB pages to allocate for Wasm execution.
+	/// Defaults to `DEFAULT_HEAP_PAGES` if `None` is provided.
+	pub fn new(wasm_method: WasmExecutionMethod, default_heap_pages: Option<u64>) -> Self {
```

Comment:
I would argue that we should have not added this argument without actually threading it to the actual configuration. Otherwise, there is a chance that we miss some places while updating

Reply:
There's only one place that a `NativeExecutor` is constructed outside of tests, so I feel like that's not much of a risk. But I'm happy to add the configuration of the Wasm method in this PR as well.

Reply:
Added a commit `cli: CLI configuration for Wasm execution method.` which does this.